spec/01-unit/08-router_spec.lua

```diff
@@ -3484,6 +3484,29 @@ for _, flavor in ipairs({ "traditional", "traditional_compatible", "expressions"
         assert.is_nil(match_t.matches.headers)
       end)
 
+      it("uri_captures works well with the optional capture group. Fix #13014", function()
+        local use_case = {
+          {
+            service = service,
+            route   = {
+              id = "e8fb37f1-102d-461e-9c51-6608a6bb8101",
+              paths = { [[~/(users/)?1984/(?<subpath>profile)$]] },
+            },
+          },
+        }
+
+        local router = assert(new_router(use_case))
+        local _ngx = mock_ngx("GET", "/1984/profile",
+                                                 { host = "domain.org" })
+        router._set_ngx(_ngx)
+        local match_t = router:exec()
+        if flavor ~= "traditional" then
+          assert.is_nil(match_t.matches.uri_captures[1])
```

Comment:
Should we use `assert.falsy()` to remove `if flavor ~= "traditional" then ... end`